src/python/dxpy/dx_extract_utils/somatic_filter_payload.py

```diff
@@ -0,0 +1,227 @@
+    # This could be the case when "--additional-fields" flag is used
+    filter_key = column_conversion.get(friendly_name,"variant_read_optimized${}".format(friendly_name))
+    # Get the condition ofr this field
+    condition = column_conditions.get(friendly_name,"in")
+    
+    listed_filter = {filter_key: [{"condition": condition, "values": values}]}
+    return listed_filter
+
+
+def location_filter(raw_location_list):
+    """
+    The somatic assay model does not currently support geno bins
+    Locations are implemented as filters on chromosome and starting position
+    Returns structure of the form:
+    {
+    "logic":"or",
+    "compound":{
+        "logic":"and"
+        "filters":{
+            "variant_read_optimized$CHROM":[
+                    {
+                    "condition":"is",
+                    "values":"12"
+                    }
+                ]
+            },
+            "variant_read_optimized$POS":[
+                {
+                "condition":"greater-than",
+                "values":"1000"
+                },
+                {
+                "condition":"less-than",
+                "values":"5000"
+                }
+            ]
+        }
+    }
+    """
+
+    # Location filters are related to each other by "or"
+    location_compound = {"compound": [], "logic": "or"}
+    for location in raw_location_list:
+        # atomic filters within an individual location filters are related by "and"
+        indiv_loc_filter = {"filters": {}, "logic": "and"}
+        start = int(location["starting_position"])
+        end = int(location["ending_position"])
+        if end - start > 250000000:
+            err_exit(
+                "Error in location {}\nLocation filters may not specify regions larger than 250 megabases".format(
+                    location
+                )
+            )
+        # First make the chr filter
+        indiv_loc_filter["filters"]["variant_read_optimized$CHROM"] = [
+            {"condition": "is", "values": location["chromosome"]}
+        ]
+        # Then the positional filters
+        indiv_loc_filter["filters"]["variant_read_optimized$POS"] = [
+            {"condition": "greater-than", "values": start},
+            {"condition": "less-than", "values": end},
+        ]
+        location_compound["compound"].append(indiv_loc_filter)
+
+    return location_compound
+
+
+def generate_pheno_filter(
+    full_input_dict, name, id, project_context, genome_reference, include_normal=False
+):
+    """
+    Generate asasy filter consisting of a compound that links the Location filters if present
+    to the regular filters
+    {
+        "pheno_filters": {
+            "id": "<id>",
+            "name":"<name>",
+            "compound":[
+                {
+                    <contents of location compound>
+                }
+            ]
+        }
+    }
+    """
+    pheno_filter = {
+        "pheno_filters": {"name": name, "id": id, "logic": "and", "compound": []}
```

Comment:
Tomorrow (6/27) we will push to vizserver the update, which will allow to send the filters as `assay_filters` and not `pheno_filters`. However, the change should land on production only on 7/11. Please update the code dependent on the release plan for this feature.
cc: @keegankelsey

Reply:
@JoeMulka can this be changed to use assay_filters properly? The vizserver support is planned to go on production next Tuesday 7/11 so definitely before CLISAM release.